src/main/java/org/smartregister/pathevaluator/task/TaskHelper.java

```diff
@@ -45,7 +46,14 @@ public void generateTask(Resource resource, Action action, String planIdentifier
 		task.setExecutionEndDate(getDateTime(action.getTimingPeriod(), false));
 		task.setAuthoredOn(DateTime.now());
 		task.setLastModified(DateTime.now());
-		task.setBusinessStatus("Not Visited");
+		for (DynamicValue dynamicValue : action.getDynamicValue()) {
+			if (dynamicValue != null && dynamicValue.getExpression() != null &&
+					dynamicValue.getExpression().getName().equals("defaultBusinessStatus")) {
+				task.setBusinessStatus(dynamicValue.getExpression().getExpression());
+			} else {
+				task.setBusinessStatus("Not Visited");
```

Comment:
This should be set after the for loop if task.getBusinessStatus is null

Reply:
Updated